### ISSUE TITLE
Problem: upload DIP to ArchivesSpace not working

### ISSUE THREAD

**Issue by kellyannewithane** [artefactual/archivematica#1112]
When try to pair a resource in ArchivesSpace with a DIP in AM the object and associated metadata does not show up in ArchivesSpace.  

Archivematica log files show that even though a match was made, the match is ignored or skipped and no information is sent to ArchivesSpace.

**Comment by mamedin** [comment]
The database shows that the file was paired:
```
select * from ArchivesSpaceDIPObjectResourcePairing;
+----+--------------------------------------+--------------------------------------+---------------------------------------+
| pk | dipUUID                              | fileUUID                             | resourceId                            |
+----+--------------------------------------+--------------------------------------+---------------------------------------+
...
...
|  6 | 3ee9a333-29e7-42c6-97b9-813ddefdc65e | 9b3c4df7-eccd-45d0-b435-8144bac3d2a4 | /repositories/3/archival_objects/9036 |
+----+--------------------------------------+--------------------------------------+---------------------------------------+
```

But the MCPClient.log says:
```
WARNING   2018-05-30 17:36:19  __main__:upload-archivesspace:upload_to_archivesspace:79:  Skipping file /var/archivematica/sharedDirectory/watchedDirectories/uploadDIP/steelhead5-3ee9a333-29e7-42c6-97b9-813ddefdc65e/objects/9b3c4df7-eccd-45d0-b435-8144bac3d2a4-MARBLES.jpg (9b3c4df7-eccd-45d0-b435-8144bac3d2a4) - no pairing found
```

**Comment by sallain** [comment]
This issue was moved to archivematica/Issues#2

**Comment by nickwilkinson** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Artefactual (internal)](https://trello.com/c/AxNvL2Bv/11-artefactual-internal)

**Comment by sromkey** [comment]
@nickwilkinson adding you to the assignee list because this requires a dev to continue work.

**Comment by jrwdunham** [comment]
@sromkey [AM PR 1177](https://github.com/artefactual/archivematica/pull/1177) contains the currently most up-to-date work on this issue. If it needs to be addressed before I return in September, that is where another dev should start looking. As I write in that PR, the current problem with the approach being explored by that PR is:

> There is at least one problem with this approach, at least as it is currently implemented: the replacement dic from the first link only affects the next link. This means that the Upload to ArchivesSpace (one instance) link lacks the AS config and will make a failed request.

**Comment by sevein** [comment]
@sromkey, these are the models/tables where the pairs are stored. It's probably the best way to inspect them:

- `AtkDIPObjectResourcePairing` (Archivists' Toolkit) (dbtable=AtkDIPObjectResourcePairing)
- `ArchivesSpaceDIPObjectResourcePairing` (ArchivesSpace) (dbtable=ArchivesSpaceDIPObjectResourcePairing)

**Comment by sevein** [comment]
The problem described in this issue is probably also affecting the integration with Archivists' Toolkit. 
However my fix only targets ArchivesSpace, but I've added a comment in the code so we can easily apply the same solution if we decide that it's needed.

    // TODO: needs to be tested with Archivists' Toolkit.

**Comment by sevein** [comment]
Ready for QA, the fix is in stable/1.7.x and qa/1.x.
